bokeh/tests/test_client_server.py

```diff
@@ -271,6 +272,45 @@ def test_io_push_to_server(self):
             # Clean up global IO state
             reset_output()
 
+    def test_client_session_callback(self):
+        application = Application()
+        with ManagedServerLoop(application) as server:
+            doc = document.Document()
+
+            client_session = ClientSession(session_id='test_client_session_callback',
+                                          url=server.ws_url,
+                                          io_loop=server.io_loop)
+            server_session = ServerSession('test_server_session_callback',
+                                            doc, server.io_loop)
+            client_session._attach_document(doc)
+
+            assert len(server_session._callbacks) == len(client_session._callbacks) == 0
```

Comment:
I know it's more verbose, but I think it's easier to debug if the test breaks if these asserts are on separate lines (the same applies below)